### DISCUSSION TITLE
Planewave source in cylindrical coordinates - Error with source definition

### DISCUSSION THREAD

**Discussion by kalosu** [NanoComp/meep#2464] [Q&A] [answered]
Hello again meep community,

I am trying to implement a simulation model similar to the one shown on https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#mie-scattering-of-a-lossless-dielectric-sphere.

I want to define a plane wave polarized along r (x) which is incident onto a microstructure from where I want to evaluate the scattered light from the structure.

Based on the tutorial, it is possible to perform this type of simulation using only one simulation for a m=-1 or 1 angular mode.

Nevertheless, once I try to run my code I get the following error message:

**meep: Not possible to place a source at r=0 in cylindrical coordinates for m = -1.**

This is how I defined the planewave source in my code:

 src = [mp.Source(src=mp.GaussianSource(fcen,fwidth=df,cutoff=10,is_integrated=True),component=mp.Er,center=      mp.Vector3(0.5*sr,0,sz*0.5-dpml),size=mp.Vector3(sr)),
mp.Source(src=mp.GaussianSource(fcen,fwidth=df,cutoff=10,is_integrated=True),component=mp.Ep,center=      mp.Vector3(0.5*sr,0,sz*0.5-dpml),size=mp.Vector3(sr),amplitude=-1j)]

where sr is the total extension of my simulation domain along r. (including the PML region)

What could be going on here? 

Notice that I have built meep from source and have used the latest codebase after a git pull.

**Comment by oskooi** [comment, marked as answer]
#2459 recently fixed a bug which had prevented placing an `Er` or `Ep` source at `r=0` for `m = ±1`. (Note, however, that it is *not* possible to place an `Ez` source at `r=0` for `m = ±1`.)

Perhaps you just need to update your build?

**Comment by kalosu** [comment]
Great, thanks!
